src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,159 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * Encapsulates the loading of tasks from a text file and storing of tasks in that text file.
+ * The text file is located in the data directory of the root folder.
+ */
+public class Storage {
+
+    /** Directory file. */
+    private final File directory;
+
+    /** Text file. */
+    private final File textFile;
+
+    /** UI for storage to display messages. */
+    private final Ui ui;
+
+    /** Task list for storage to access. */
+    private final TaskList taskList;
+
+    /**
+     * Creates a storage for Duke's tasks data.
+     *
+     * @param directoryPath Path of the directory.
+     * @param textFilePath Path of the text file.
+     * @param ui UI.
+     * @param taskList List of tasks.
+     */
+    public Storage(String directoryPath, String textFilePath, Ui ui, TaskList taskList) {
+        this.directory = new File(directoryPath);
+        this.textFile = new File(textFilePath);
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Gets the UI.
+     *
+     * @return UI.
+     */
+    public Ui getUi() {
+        return ui;
+    }
+
+    /**
+     * Gets the list of tasks.
+     *
+     * @return List of tasks.
+     */
+    public TaskList getTaskList() {
+        return taskList;
+    }
+
+    /**
+     * Checks if duke.txt file exists in the data directory of the root folder.
+     * If it exists, load the data into Duke.
+     * If it does not exist, create one.
+     */
+    public void checkSaveFile() {
+        if (directory.exists()) {
+            if (textFile.exists()) {
+                readTextFile();
+            } else {
+                createTextFile();
+            }
+        } else {
+            createDataDirectory();
+            createTextFile();
+        }
+    }
+
+    /**
+     * Creates a data directory in the root folder.
+     */
+    private void createDataDirectory() {
+        boolean hasCreatedDir = directory.mkdir();
+        if (hasCreatedDir) {
+            this.ui.printMessage("A data directory has been created in the root folder.");
+        } else {
+            this.ui.printWarning("Failed to create a data directory.");
+        }
+    }
+
+    /**
+     * Creates duke.txt in the data directory of the root folder.
+     */
+    private void createTextFile() {
```

Comment:
Perhaps call these functions `doSomethingSaveFile` rather than `doSomethingTextFile` since you are referring to the save file all along? Could reduce some confusion.

Reply:
In this case, I have to create a directory and a text file. The checkedSaveFile method checks for the existence of both the directory and save file. I think it  will be better to separate these differences. Moreover, the method is private so user will not need to interact with this method.